pkg/storage-init/storage-init.sh

```diff
@@ -34,6 +34,57 @@ fi
 
 P3_FS_TYPE="ext3"
 FSCK_FAILED=0
+
+# First lets see if we're missing P3 altogether and try to create it.
+# This is safe, since the worst case scenario we may hose the system,
+# but the system without P3 is a warm brick anyways
+P3=$(/hostfs/sbin/findfs PARTLABEL=P3)
+IMGA=$(/hostfs/sbin/findfs PARTLABEL=IMGA)
```

Comment:
This basically says, "if we don't have a `/persist` partition, and we do have IMGA, then calculate where the `/persist` should exist and make it".

If we don't have an IMGA, we probably should be panicking, so maybe let's make this cleaner and even earlier do an if statement that says, `if [ -z "$IMGA" ]; then panic`? That will make this one cleaner and easier to understand.

Also, let's add a comment here as to what the logic is, something like, "if we don't have a persist partition, calculate where it should sit and create it."

Reply:
Yes. But on the later point -- I don't think we should be  panicking actually -- I think a more sensible strategy is still to allow the box to boot into the shell. Note that whatever booted us is 99.99% a full EVE rootfs image (and yes it may not be marked as IMGA -- but it still is a full image). This means that we can leverage EVE microservices to communicate useful bits and pieces of information back to the cloud.

I fixed the comment.

Reply:
> Yes. But on the later point -- I don't think we should be panicking actually -- I think a more sensible strategy is still to allow the box to boot into the shell. Note that whatever booted us is 99.99% a full EVE rootfs image (and yes it may not be marked as IMGA -- but it still is a full image). This means that we can leverage EVE microservices to communicate useful bits and pieces of information back to the cloud.

@rvs We clearly won't be sending logs to the controller, since we have not /persist/rsyslogd.
Or would we run with /persist in memory, meaning that we can get various subsequent failures which would be hard to diagnose? I think we need to 1) have specific logic to handle the case of no /persist (not watchdog, or run out of memory due to overlayfs) 2) test that the system can operate in this space so that it can at least report to the controller and the controller can send a reboot command.